spec/spec_helper.rb

```diff
@@ -49,4 +49,32 @@
   end
 end
 
+# Takes a hash from model class to the number of expected queries executed for that model e.g.
+# expect {  }.to execute_queries(Post => 2, User => 1)
+RSpec::Matchers.define(:execute_queries) do |expected_counts|
+  match(notify_expectation_failures: true) do |actual|
+    @actual_queries = []
+    listener = lambda do |_name, _start, _finish, _message_id, values|
+      @actual_queries << values[:sql]
+    end
+
+    ActiveSupport::Notifications.subscribed(listener, 'sql.active_record', &actual)
+
+    expected_counts_by_table = expected_counts.transform_keys(&:table_name)
+
+    table_extractor = /SELECT .* FROM "(.+)" WHERE/
+    actual_counts_by_table = @actual_queries.group_by do |query|
+      table_extractor.match(query)[1]
+    end.transform_values(&:size)
+
+    actual_counts_by_table == expected_counts_by_table
```

Comment:
Would it make sense to verify the record count too ? I could imagine a bug where you are requesting extra ids by mistake. Do you even have access to it?

Reply:
That would be a useful thing to check. We have the SQL and the bind variables at this point so in theory we could try to figure that out but I'd like to avoid parsing the SQL (anymore than my hacky regex is already doing!) and becoming too dependent on how Rails is generating the SQL. I think I'll punt on that for now to get the tests passing again but will figure out a way to make it work if we ever hit a bug in this area.